### PR TITLE
add secureboot test

### PR DESCRIPTION
**Pull request by gaohannk**
image_boot git:(image-test-secureboot) /out/manager -project hanga-testing -zone us-west1-c -images projects/rhel-cloud/global/images/rhel-8-v20210420
2021/04/28 22:14:31 imagetest: Done with setup
runTestWorkflow: running image_validation on rhel-8-v20210420 (ID 331l4)
runTestWorkflow: running image_boot on rhel-8-v20210420 (ID qb0h0)
````<testsuites name="" errors="0" failures="0" tests="0" time="0">
	<testsuite name="image_validation-rhel-8-v20210420" tests="4" failures="0" errors="0" disabled="0" skipped="0" time="0">
		<testcase classname="" name="TestHostname" time="0"></testcase>
		<testcase classname="" name="TestFQDN" time="0"></testcase>
		<testcase classname="" name="TestHostKeysGeneratedOnce" time="1.42"></testcase>
		<testcase classname="" name="TestCustomHostname" time="0.01"></testcase>
	</testsuite>
	<testsuite name="image_boot-rhel-8-v20210420" tests="3" failures="0" errors="0" disabled="0" skipped="0" time="0">
		<testcase classname="" name="TestGuestBoot" time="0"></testcase>
		<testcase classname="" name="TestGuestReboot" time="0"></testcase>
		<testcase classname="" name="TestGuestSecureBoot" time="30"></testcase>
	</testsuite>
</testsuites>
```

### CODE REVIEW

**Comment by gaohannk** [comment]
Comment out setup to not enable secure boot and rebuild the manager.
//vm2.EnableSecureBoot()
rerun test: 
```
2021/04/29 18:00:21 imagetest: Done with setup
runTestWorkflow: running image_boot on rhel-8-v20210420 (ID 6rwpp)
runTestWorkflow: running image_validation on rhel-8-v20210420 (ID 97gdm)
<testsuites name="" errors="0" failures="0" tests="0" time="0">
	<testsuite name="image_validation-rhel-8-v20210420" tests="4" failures="0" errors="0" disabled="0" skipped="0" time="0">
		<testcase classname="" name="TestHostname" time="0.02"></testcase>
		<testcase classname="" name="TestFQDN" time="0.02"></testcase>
		<testcase classname="" name="TestHostKeysGeneratedOnce" time="0.11"></testcase>
		<testcase classname="" name="TestCustomHostname" time="0.01"></testcase>
	</testsuite>
	<testsuite name="image_boot-rhel-8-v20210420" tests="3" failures="1" errors="0" disabled="0" skipped="0" time="0">
		<testcase classname="" name="TestGuestBoot" time="0"></testcase>
		<testcase classname="" name="TestGuestReboot" time="0"></testcase>
		<testcase classname="" name="TestGuestSecureBoot" time="30">
			<failure type="">    image_boot_test.go:48: secure boot is not enabled as expected</failure>
		</testcase>
	</testsuite>
</testsuites>
```

**Review by hopkiw — CHANGES_REQUESTED** [review]

File: imagetest/fixtures.go [hunk 1]
```diff
@@ -120,3 +121,12 @@ func (t *TestVM) Reboot() error {
 
 	return nil
 }
+
+// EnableSecureBoot make all test VMs in workflow with secure boot.
+func (t *TestVM) EnableSecureBoot() {
+	for _, i := range t.testWorkflow.wf.Steps[createVMsStepName].CreateInstances.Instances {
```
[the remark this answers is not in the document]
Comment (gaohannk) [reply imagetest/fixtures.go:127]: Good Catch, should be the one that call this method.

File: imagetest/test_suites/image_boot/setup.go [hunk 2]
```diff
@@ -11,5 +11,17 @@ func TestSetup(t *imagetest.TestWorkflow) error {
 	if err != nil {
 		return err
 	}
-	return vm.Reboot()
+	if err := vm.Reboot(); err != nil {
+		t.Skip("reboot workflow failed")
```
Comment (hopkiw) [inline imagetest/test_suites/image_boot/setup.go:15]: oh this should not be a skip. skip is for when you want to disable a test suite on some condition. in this case, just return the err

File: imagetest/test_suites/image_boot/image_boot_test.go [hunk 3]
```diff
@@ -37,3 +44,20 @@ func TestGuestReboot(t *testing.T) {
 	// second boot
 	t.Log("marker file exist signal the guest reboot successful")
 }
+
+func TestGuestSecureBoot(t *testing.T) {
+	// UEFI vars are (apparently) not immediately available, but sleeping for
```
[the remark this answers is not in the document]
Comment (gaohannk) [reply imagetest/test_suites/image_boot/image_boot_test.go:49]: https://wiki.archlinux.org/index.php/Unified_Extensible_Firmware_Interface#UEFI_variables_support_in_Linux_kernel

File: imagetest/fixtures.go [hunk 4]
```diff
@@ -120,3 +121,14 @@ func (t *TestVM) Reboot() error {
...
 	return nil
 }
+
+// EnableSecureBoot make the current test VMs in workflow with secure boot.
+func (t *TestVM) EnableSecureBoot() {
+	for _, i := range t.testWorkflow.wf.Steps[createVMsStepName].CreateInstances.Instances {
+		if i.Name == t.name {
+			i.ShieldedInstanceConfig = &compute.ShieldedInstanceConfig{
+				EnableSecureBoot: true,
+			}
```
Comment (hopkiw) [inline imagetest/fixtures.go:131]: add 'break'

File: imagetest/test_suites/image_boot/image_boot_test.go [hunk 5]
```diff
@@ -37,3 +41,17 @@ func TestGuestReboot(t *testing.T) {
 	// second boot
 	t.Log("marker file exist signal the guest reboot successful")
 }
+
+func TestGuestSecureBoot(t *testing.T) {
+	if _, err := os.Stat(secureBootFile); os.IsNotExist(err) {
+		t.Skip("not supported on non-uefi boot disk")
```
Comment (hopkiw) [inline imagetest/test_suites/image_boot/image_boot_test.go:47]: it is not a skip, it is a fail

Comment (hopkiw) [reply imagetest/test_suites/image_boot/image_boot_test.go:47]: to elaborate, if this test were run against all VMs, skip would be fine here. but we specifically set up a VM to run this test against, and if the entire sysfs var is missing, that's a fail.

Comment (gaohannk) [reply imagetest/test_suites/image_boot/image_boot_test.go]: I assume the test run against al lvm. but it's good to consider the case that efi var is missing.

**Pull request merged**